httpbp/healthcheck.go

```diff
@@ -0,0 +1,62 @@
+package httpbp
+
+import (
+	"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/reddit/baseplate.go/internal/gen-go/reddit/baseplate"
+)
+
+// HealthCheckProbeQuery is the name of HTTP query defined in Baseplate spec.
+const HealthCheckProbeQuery = "type"
+
+// GetHealthCheckProbe parses the health check probe from the request.
+//
+// Unrecognized string probes will fallback to READINESS.
+// When that happens a non-nil error will also be returned.
+// If a probe is not specified in the request,
+// this function will return READINESS with nil error.
+//
+// This function only supports the probes known to this version of Baseplate.go
+// for the string version of probes passed into the request.
+// If in the future a new probe is added to baseplate.thrift,
+// you would need to update Baseplate.go library,
+// otherwise it would fallback to READINESS.
+// Currently the supported probes are:
+//
+// - READINESS
+//
+// - LIVENESS
+//
+// - STARTUP
+//
+// If the probe specified in the request is the int value,
+// this function just blindly return the parsed int value,
+// even if it's not one of the defined enum values known to this version of
+// Baseplate.go.
+//
+// Note that because of go's type system,
+// to make this function more useful the returned probe value is casted back to
```

Comment:
Does that _actually_ make it more useful? It means that if you want to compare it to the actual enum values you have to convert it _back_?

Reply:
So if we just return the thrift type, it would be actually `github.com/reddit/baseplate.go/internal/gen-go/reddit/baseplate.IsHealthyProbe` type. Services don't really have access to that type (it's under `internal/`), and their own thrift compiled type would have a different import path and they need a typecasting anyways.

Reply:
Right, I keep forgetting that 🤦